src/spinegeneric/cli/check_data_consistency.py

```diff
@@ -56,70 +103,39 @@ def main():
+        for subj in sorted(tsv_subj - dir_subj):
+            print(f"Warning: participants.tsv: missing data folder for subject '{subj}'")
 
     # Check the presence of JSON sidecars
     for img_path in args.path_in.glob("sub-*/**/*.nii.gz"):
         json_path = img_path.with_suffix("").with_suffix(".json")
         if not json_path.exists():
-            print(f"Missing jsonSidecar: {json_path}")
-
-    # Checking participants.tsv contents
-    schema = Schema(
-        [
-            Column(
-                "participant_id",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column("sex", [InListValidation(["M", "F"])]),
-            Column("age", [InRangeValidation(18, 60)]),
-            Column("height", [MatchesPatternValidation(r"[0-9]|-")]),
-            Column("weight", [MatchesPatternValidation(r"[0-9]|-")]),
-            Column(
-                "date_of_scan",
-                [DateFormatValidation("%Y-%m-%d") | MatchesPatternValidation(r"-")],
-            ),
-            Column(
-                "institution_id",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column(
-                "institution",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column(
-                "manufacturer",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column(
-                "manufacturers_model_name",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column(
-                "receive_coil_name",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column(
-                "software_versions",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-            Column(
-                "researcher",
-                [LeadingWhitespaceValidation(), TrailingWhitespaceValidation()],
-            ),
-        ]
-    )
-
-    tsv_file = pd.read_csv(str(args.path_in / "participants.tsv"), sep="\t")
-    errors = schema.validate(tsv_file)
-    print("\nChecking the contents of participants.tsv")
-    if not errors:
-        print("--> all good 👍")
-    else:
-        for error in errors:
-            print(error)
+            print(f"Warning: missing JSON sidecar for {img_path}")
+
+    # Check the contents of participants.tsv
+    tsv_cols = set(fieldnames)
+    expected_cols = set(validators.keys())
+    for col in sorted(expected_cols - tsv_cols):
+        print(f"Warning: participants.tsv: missing column '{col}'")
+    for col in sorted(tsv_cols - expected_cols):
+        print(f"Warning: participants.tsv: extra column '{col}'")
+    for r, row in enumerate(rows, start=1):
```

Comment:
Tiny organizational nit to avoid a "wall of text" feel:

```suggestion
    # Check the column names of `participants.tsv` to ensure they match the validator
    tsv_cols = set(fieldnames)
    expected_cols = set(validators.keys())
    for col in sorted(expected_cols - tsv_cols):
        print(f"Warning: participants.tsv: missing column '{col}'")
    for col in sorted(tsv_cols - expected_cols):
        print(f"Warning: participants.tsv: extra column '{col}'")
        
    # Check the row values of `participants.tsv` using the validator
    for r, row in enumerate(rows, start=1):
```

Reply:
Thanks! Done in https://github.com/spine-generic/spine-generic/pull/287/commits/34cf0daead6c71d1f3b16867da4564171aed0519.